tide-whisperer.go

```diff
@@ -167,6 +147,39 @@ func main() {
 		log.Printf("%s mongo processing finished after %.5f seconds and returned %d records", DATA_API_PREFIX, time.Now().Sub(startTime).Seconds(), writeCount)
 	}
 
+	authorize := func(h http.Handler) http.Handler {
+		return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+
+			start := time.Now()
+
+			token := r.Header.Get("x-tidepool-session-token")
+			accessTokenHeader := r.Header.Get("Authorization")
+			if len(accessTokenHeader) > 7 && strings.EqualFold(accessTokenHeader[0:7], "BEARER ") {
```

Comment:
Will we ever have an `Authorization` header that *doesn't* meet these guidelines. If not, I'd get rid of this check entirely and replace it with `if accessTokenHeader != ""`.

Also, I'd put the the Get of X-Tidepool-Session-Header after if there is no token. So:

```
var token string
if authorization := r.Header.Get("Authorization"); authorization != "" {
    if parts := strings.Split(authorization, " "); len(parts) == 2 && strings.EqualFold(parts[0], "Bearer") {
        log.Println("Validating with access_token")
        token = parts[1]
    }
}
if token == "" {
    token = r.Header.Get("X-Tidepool-Session-Token")
}
```